src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -207,30 +231,39 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
 
             foreach (var emulatorResource in eventHubsEmulatorResources)
             {
-                var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().Single(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
+                // A custom file mount with read-only access is used to mount the emulator configuration file. If it's not found, the read-write mount we defined on the container is used.
+                var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().LastOrDefault(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
 
+                // If the latest mount for EmulatorConfigJsonPath is our custom one then we can generate it.
+                if (configFileMount != customMountAnnotation)
+                {
+                    continue;
+                }
                 using var stream = new FileStream(configFileMount.Source!, FileMode.Create);
-                using var writer = new Utf8JsonWriter(stream);
+                using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions { Indented = true });
+
+                if (!OperatingSystem.IsWindows())
+                {
+                    File.SetUnixFileMode(configHostFile,
+                        UnixFileMode.UserRead | UnixFileMode.UserWrite
+                        | UnixFileMode.GroupRead | UnixFileMode.GroupWrite
+                        | UnixFileMode.OtherRead | UnixFileMode.OtherWrite);
+                }
 
                 writer.WriteStartObject();                      // {
                 writer.WriteStartObject("UserConfig");          //   "UserConfig": {
                 writer.WriteStartArray("NamespaceConfig");      //     "NamespaceConfig": [
                 writer.WriteStartObject();                      //       {
-                writer.WriteString("Type", "EventHub");         //         "Type": "EventHub",
+                writer.WriteString("Type", "EventHub");
 
                 // This name is currently required by the emulator
-                writer.WriteString("Name", "emulatorNs1");      //         "Name": "emulatorNs1"
+                writer.WriteString("Name", "emulatorNs1");
                 writer.WriteStartArray("Entities");             //         "Entities": [
 
                 foreach (var hub in emulatorResource.Hubs)
                 {
-                    // The default consumer group ('$default') is automatically created
-
-                    writer.WriteStartObject();                  //           {
-                    writer.WriteString("Name", hub);            //             "Name": "hub",
-                    writer.WriteString("PartitionCount", "2");  //             "PartitionCount": "2",
```

Comment:
We used to have "2" as the default, but `hub.WriteJsonObjectProperties` now writes "1" by default.

Is this intentional?

Reply:
AFAIR it was 2 because of the sample they provided. But since the schema states [1-32] I decided to use 1 when the value is not set explicitly.